src/test/scala/mesosphere/marathon/core/appinfo/TaskCountsTest.scala

```diff
@@ -0,0 +1,132 @@
+package mesosphere.marathon.core.appinfo
+
+import mesosphere.marathon.health.Health
+import mesosphere.marathon.state.Timestamp
+import mesosphere.marathon.{ MarathonSpec, Protos }
+import mesosphere.util.Mockito
+import org.apache.mesos.{ Protos => mesos }
+import org.scalatest.{ GivenWhenThen, Matchers }
+
+class TaskCountsTest extends MarathonSpec with GivenWhenThen with Mockito with Matchers {
+  test("count no tasks") {
+    When("getting counts for no tasks")
+    val counts = TaskCounts(appTasks = Seq.empty, statuses = Map.empty)
+    Then("all counts are zero")
+    counts should be(TaskCounts.zero)
+  }
+
+  test("one unstaged task") {
+    Given("one unstaged task")
+    val oneUnstagedTask = Seq(
+      unstagedTask("task1")
+    )
+    When("getting counts")
+    val counts = TaskCounts(appTasks = oneUnstagedTask, statuses = Map.empty)
+    Then("all counts are 0")
+    counts should be(TaskCounts.zero)
+  }
+
+  test("one staged task") {
+    Given("one staged task")
+    val oneStagedTask = Seq(
+      stagedTask("task1")
+    )
+    When("getting counts")
+    val counts = TaskCounts(appTasks = oneStagedTask, statuses = Map.empty)
+    Then("all counts are 0 except staged")
+    counts should be(TaskCounts.zero.copy(tasksStaged = 1))
+  }
+
+  test("one running task") {
+    Given("one running task")
+    val oneRunningTask = Seq(
+      runningTask("task1")
+    )
+    When("getting counts")
+    val counts = TaskCounts(appTasks = oneRunningTask, statuses = Map.empty)
+    Then("all counts are 0 except running")
+    counts should be(TaskCounts.zero.copy(tasksRunning = 1))
+  }
+
+  test("one healthy task") {
+    Given("one unstaged task with alive Health")
+    val oneRunningTask = Seq(
+      unstagedTask("task1")
+    )
+    When("getting counts")
+    val counts = TaskCounts(appTasks = oneRunningTask, statuses = Map("task1" -> aliveHealth))
+    Then("all counts are 0 except healthy")
+    counts should be(TaskCounts.zero.copy(tasksHealthy = 1))
+  }
+
+  test("one unhealthy task") {
+    Given("one unstaged task with !alive health")
+    val oneRunningTask = Seq(
+      unstagedTask("task1")
+    )
+    When("getting counts")
+    val counts = TaskCounts(appTasks = oneRunningTask, statuses = Map("task1" -> notAliveHealth))
+    Then("all counts are 0 except tasksUnhealthy")
+    counts should be(TaskCounts.zero.copy(tasksUnhealthy = 1))
+  }
+
+  test("a task with mixed health is counted as unhealthy") {
+    Given("one task with mixed health")
+    val oneRunningTask = Seq(
+      unstagedTask("task1")
```

Comment:
same as above – although `TaskCounts`doesn't consider the task status for health stats, the test should make sense. (Or maybe the test should fail because an unstaged task should actually be considered neither healthy nor unhealthy.)